### PR TITLE
feat(docs): add `embedded typegate` page

### PR DESCRIPTION
**Pull request by destifo**
# Embedded typegate docs page

- [x] need a page under meta-cli/embedded typegate to explain how that works
- [x] tutorials should take advantage of the embedded one
- [x] the embedded one should be the default everything
- [x] explain that there is 2 flavors in reference/meta-cli

[MET-562](https://linear.app/metatypedev/issue/MET-562/docs-use-embedded-whenever-possible-in-the-docs-and-examples)

-

-

#### Migration notes

_No Migrations Needed_
...

- [ ] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: website/shared/meta-cli/embedded-typegate.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+If you have installed the _non-thin_ version of _Meta CLI_, it comes with an embedded typegate packaged inside it. The [typegate](/docs/reference/typegate) is where we deploy our [typegraphs](/docs/reference/typegraph) where any logic written in them is exposed via an HTTP or GraphQL endpoints. You can run an embedded typegate node from the terminal, but you first need to set a couple of environment variables which are needed to instantiate the typegate node.
```
Comment (zifeo) [inline website/shared/meta-cli/embedded-typegate.mdx:2]: double where, let's maybe rephrase?

File: website/shared/meta-cli/embedded-typegate.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+If you have installed the _non-thin_ version of _Meta CLI_, it comes with an embedded typegate packaged inside it. The [typegate](/docs/reference/typegate) is where we deploy our [typegraphs](/docs/reference/typegraph) where any logic written in them is exposed via an HTTP or GraphQL endpoints. You can run an embedded typegate node from the terminal, but you first need to set a couple of environment variables which are needed to instantiate the typegate node. 
+
+Set the `tg_admin_password` and `tg_secret` environment variables. You can use the following command to configure a sample value for the variables and test the embedded typegate.
+```shell
+// set up tg_secret and tg_admin_password
+export tg_secret=a4lNi0PbEItlFZbus1oeH/+wyIxi9uH6TpL8AIqIaMBNvp7SESmuUBbfUwC0prxhGhZqHw8vMDYZAGMhSZ4fLw== | export tg_admin_password=password
```
Comment (zifeo) [inline website/shared/meta-cli/embedded-typegate.mdx:7]: is the pipe really required here?

File: website/shared/meta-cli/embedded-typegate.mdx (hunk 1 above)
Comment (zifeo) [inline website/shared/meta-cli/embedded-typegate.mdx:2]: I would assume that they have by default, and instead of highlight the non-thin, the thin flavor shall instead of the "exception"

File: website/shared/meta-cli/embedded-typegate.mdx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+```
+
+If you have not installed _Meta_CLI_ or you have downloaded the _thin_ version, you can check [this](/docs/reference/meta-cli#Installation) installation guide of the _CLI_.
+
+Everything is setup to run the embedded typegate. Just run the following command below.
```
Comment (zifeo) [inline website/shared/meta-cli/embedded-typegate.mdx:12]: does metatype dev spawn the type gate automatically?

File: website/shared/projects/tab-first-project-python.mdx [hunk 4]
```diff
@@ -97,8 +97,11 @@ With these simple steps, we were able to build a basic backend with database cap
   {require("../../../examples/typegraphs/quick-start-project.py").content}
 </CodeBlock>
 
-We are almost there to test your first `Metatype` application. We now need to spin a [Tyepgate](/docs/reference/typegate) and deploy our typegraph to the instance. To start the typegate instance, run the following command.
+We are almost there to test your first `Metatype` application. We now need to spin a [Tyepgate](/docs/reference/typegate) and deploy our typegraph to the instance. We have two options to start a typegate instance. The first and the recommended one is to use the [embedded typgate](/docs/reference/meta-cli/embedded-typegate) that comes with the _non-thin_ version of _Meta_CLI_. The second option is to run a docker container from our `compose.yml` file.
 
+To run the embedded typegate, you can follow this [reference](/docs/reference/meta-cli/embedded-typegate). You this option is preferrable as it spins a simple typegate for development.
```
Comment (zifeo) [inline website/shared/projects/tab-first-project-python.mdx:102]: let's not give a choice on the first time user, it shall be as simple as possible so no choice should be left, they can lean about the other options later

File: website/shared/projects/tab-first-project-python.mdx [hunk 5]
```diff
@@ -97,8 +97,11 @@ With these simple steps, we were able to build a basic backend with database cap
   {require("../../../examples/typegraphs/quick-start-project.py").content}
 </CodeBlock>
 
-We are almost there to test your first `Metatype` application. We now need to spin a [Tyepgate](/docs/reference/typegate) and deploy our typegraph to the instance. To start the typegate instance, run the following command.
+We are almost there to test your first `Metatype` application. We now need to spin a [Tyepgate](/docs/reference/typegate) and deploy our typegraph to the instance. We have two options to start a typegate instance. The first and the recommended one is to use the [embedded typgate](/docs/reference/meta-cli/embedded-typegate) that comes with the _non-thin_ version of _Meta_CLI_. The second option is to run a docker container from our `compose.yml` file.
```
Comment (zifeo) [inline website/shared/projects/tab-first-project-python.mdx:100]: If I recall Diátaxis, shouldn't "we" avoid the "we"? and speaking instead to the user directly?

**Comment by zifeo** [comment]
Nice improvements! A few phrasing suggestions :)

File: website/shared/meta-cli/embedded-typegate.mdx (hunk 3 above)
Comment (destifo) [reply website/shared/meta-cli/embedded-typegate.mdx:12]: I am on version 0.3.6 and `meta dev` doesn't spawn the typegate automatically.

**Review by Yohe-Am — APPROVED** [review]

**Review by zifeo — APPROVED** [review]
Nice rewordings!
